### DISCUSSION TITLE
Is there a way to include VML in my email templates?

### DISCUSSION THREAD

**Discussion by dilly-sixmo** [supabase/supabase#14065] [Questions] [answered]
I've crafted a lovely looking email that not only looks great in the browser, but in Outlook as well.  This required the use of VML as we all know Outlook is notoriously bad when it comes to CSS.  That means, within my email template HTML code, I have sections like this interspersed:

```
        <!--[if gte mso 9]>
                <v:rect xmlns:v="urn:schemas-microsoft-com:vml" fill="true" stroke="false" style="width:806px;height:38px;">
<v:fill type="tile" src="https://something.supabase.co/storage/v1/object/public/images/image.png" color="#ffffff" />
<v:textbox inset="0,0,0,0">
<![endif]-->
        <div>
        </div>
        <!--[if gte mso 9]>
                 </v:textbox>
             </v:rect>
             <![endif]--
```

The problem is that when I save my email template code, this portion of the template code gets automatically removed.  This is a problem.  I need this code in my template.  I looked around the database to see if I could find where these email templates were stored as I figured I would just edit the database table manually and force it to accept my VML code!  But unfortunately I was not able to locate any table that contains the email template code I created.  Apparently they are stored elsewhere.

I know I can set up a custom SMTP server, but that doesn't help me, as I'm still left sending the templates that I made within Supabase for things like password resets, etc.

I know this is a part of Supabase that will be updated in the future, but I'm looking for a solution in the interim.  Is there any way I can get the Supabase Email Template editor to accept my VML code?

**Comment by silentworks** [comment, marked as answer]
At the moment the only workaround would be to send the email from your own emailing solution. You can generate the URL for any of the auth functionalities using the [`.admin.generateLink`](https://supabase.com/docs/reference/javascript/auth-admin-generatelink) method of the supabase-js library.
```ts
const { data, error } = await supabase.auth.admin.generateLink({
  type: 'signup',
  email: 'email@example.com',
  password: 'secret'
})
``` 
This way you could potentially trigger a webhook that calls the edge function to send a email.

**Reply by dilly-sixmo** [reply]
Thanks for bringing this method to my attention as I was not aware of it.  I think I'll be able to accomplish what I'm after now.
